src/targets/node/nodeAttacher.ts

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import { AnyLaunchConfiguration, INodeAttachConfiguration } from '../../configuration';
+import { Contributions } from '../../common/contributionUtils';
+import { getWSEndpoint } from '../browser/launcher';
+import { spawnWatchdog } from './watchdogSpawn';
+import { NodeLauncherBase, IRunData } from './nodeLauncherBase';
+import { findInPath } from '../../common/pathUtils';
+import { Socket } from 'net';
+import { SubprocessProgram } from './program';
+
+/**
+ * Attaches to ongoing Node processes. This works pretty similar to the
+ * existing Node launcher, except with how we attach to the entry point:
+ * we don't have the bootloader in there, so we manually attach and enable
+ * the debugger, then evaluate and set the environment variables so that
```

Comment:
Not sure we should set environment variables when we attach. The existing DA doesn't work that way and it could cause strange behavior because they will be set at some random point in the program's execution

Reply:
The downside is that not doing so will cause any child processes to not be instrumented. I think having an option for this would be good, I'll make a note to do that when I implement it.

Reply:
Something I also talked about with Kai yesterday was having an option to aggressively attach to any existing child processes of the tree (in an option). We already have the logic needed to get the process tree, if the children are run with `--inspect` we can connect to them out of the box, otherwise we could try to attempt to `putPidInDebugMode`. That's not for this PR, but could be a useful extension for scenarios where people run a cluster of long-running processes.

Reply:
Which env var controls that, NODE_OPTIONS? I agree that we should set ones for the debugger. Not user specified ones.

That's a good idea, there is also the 'cluster' module which will spawn children in debug mode if the parent is in debug mode. We already support that, we should support it here too.

Reply:
Yep, NODE_OPTIONS, and then we have some extra options for the watchdog. I wouldn't touch any of the extra/user-specified variables.